### DISCUSSION TITLE
How to include app.js & app.css from resources without changing base templates

### DISCUSSION THREAD

**Discussion by sergeyzozulya** [filamentphp/filament#4507] [Help]
Hi everyone,

Let's say we have a simple app created with something like this:

```
composer create-project laravel/laravel test
composer require filament/filament:"^2.0"

```

Easy and simple, everything works out of the box until you decide to update app.js or app.css from resources/js and resources/css folders. It turns out that those files are not connected. I do understand that Laravel requires using webpack.mix.js to compile everything but from the other side Filament provides own layouts and own vendor/filament/filament/resources/js/app.js

So my question is how to add custom JS to default Filament Admin?

Thanks in advance for any help

Sergey

**Comment by Kristories** [comment]
Register your own scripts and styles using the `registerScripts()` and `registerStyles()` methods in a service provider's `boot()` method:

```
use Filament\Facades\Filament;
 
Filament::registerScripts([
    asset('js/my-script.js'),
]);
 
Filament::registerStyles([
    'https://unpkg.com/tippy.js@6/dist/tippy.css',
    asset('css/my-styles.css'),
]);
```

Ref : https://filamentphp.com/docs/2.x/admin/appearance#including-frontend-assets

**Reply by sergeyzozulya** [reply]
This way we add reference to JS to output HTML, that's fine
But Laravel 9 and Vite don't "build" mentioned resources/js/app.js to public/build
Instead vendor/filament/filament/resources/js/app.js is being built
I would assume that Laravel 8 would work fine but I would prefer Laravel 9
Any solution?

**Reply by sergeyzozulya** [reply]
Official doc: Vite lacks some features that Laravel Mix supports, such as the ability to copy arbitrary assets into the build that are not referenced directly in your JavaScript application.

This means that in default configuration there is no chance for resources/js/app.js to come to public/build
And reverting back to mix won't be a solution
